Cython/Compiler/TypeStubGenerator.py

```diff
@@ -0,0 +1,433 @@
+from . import Version
+from .Nodes import CNameDeclaratorNode
+from .ExprNodes import CallNode, NameNode, ImportNode, TupleNode, AttributeNode
+from ..CodeWriter import DeclarationWriter
+from .Visitor import CythonTransform
+from . import PyrexTypes
+from ..Utils import open_new_file
+import cython 
+import os 
+import sys 
+
+cython.declare(PyrexTypes=object, Naming=object, ExprNodes=object, Nodes=object,
+               Options=object, UtilNodes=object, LetNode=object,
+               LetRefNode=object, TreeFragment=object, EncodedString=object,
+               error=object, warning=object, copy=object, _unicode=object)
+
+
+# Inspired by and based around https://github.com/cython/cython/pull/3818
+# with some less lazy changes to it and a few minor improvements and optimizations...
+
+# Decided to revert to an older variant I had wrote of this code for the sake of 
+# maintainability - Vizonex
+
+
+
+# TODO Save this implementation commented out if required....
+if sys.version_info >= (3, 9):
+    typing_module = "typing"
+else:
+    typing_module = "typing_extensions"
+
+class PyiWriter(CythonTransform, DeclarationWriter):
+    """Used By Cython to help Write stubfiles
+    this comes in handy for ides like Pylance 
+    which suffer from having no code access to 
+    annotations from compiled python modules...
+    """
+
+    def __init__(self, context):
+        super(PyiWriter, self).__init__(context=context)
+        super(DeclarationWriter, self).__init__()
+        self.context = context
+        self.module_name = ""
+        self.class_func_count = 0 
+
+        self.translation_table = {}
+        """Used as an eternal resource for translating ctype declarations into python-types"""
+
+        self.use_typing = False
+        """if true we must import typing's generator typehint..."""
+
+
+    def _visitchildren_indented(self, node):
+        self.indent()
+        self.visitchildren(node)
+        self.dedent()
+    
+    def translate_pyrex_type(self, ctype):
+        # TODO implement Pyrex to cython shadow typehints converter...
+
+        if isinstance(ctype, PyrexTypes.BuiltinObjectType):
+            return ctype.py_type_name()
+
+        if isinstance(ctype, PyrexTypes.CVoidType):
```

Comment:
Most of these are defined as global constants in PyrexTypes. This one you can test with

```
if ctype is c_void_type:
```